src/refactor_bot/agents/test_validator.py

```
@@ -40,17 +42,110 @@ def __init__(
+        self.llm_fallback_provider = (
+            self._normalize_provider(llm_fallback_provider)
+            if llm_fallback_provider
+            else None
+        )
+        self.allow_fallback = bool(allow_fallback)
+        self.allow_human_fallback = bool(allow_human_fallback)
+        if self.allow_fallback and self.llm_fallback_provider:
+            if self.llm_fallback_provider == "anthropic" and self._anthropic_client is None:
+                raise TestValidationError(
+                    "Fallback provider requested as anthropic but ANTHROPIC_API_KEY is not set."
+                )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)